### PR TITLE
imshow for nd image

### PR DESCRIPTION
**Pull request by johnnychen94**
Thanks for @mcabbott making the proposal in #21. This PR fixes some issues there. (closes #21)

Under JuliaImages, `Array{T, 3}` image is interpreted as a 3d image by default. There are several ways to slicing it, I choose to slice images along dims 3. Not very sure if we really need to expose the `dims` keyword argument.

Preview:

```julia
julia> Gray.(rand(2,16, 3))
2×16×3 Array{Gray{Float64},3} with eltype Gray{Float64}:
slice: 1/3
████████████████████████████████
████████████████████████████████
slice: 2/3
████████████████████████████████
████████████████████████████████
slice: 3/3
████████████████████████████████
████████████████████████████████
```

Will add test cases later.

### CODE REVIEW

**Comment by mcabbott** [comment]
Thanks for taking a look, this works too! 

The two advantages of #21's method are that it prints `[:, :, 1] =` below `summary`, things I already understand, and that if you accidentally run this it only prints a few pages worth, not all of them:
```
julia> Gray.(rand(100,100,100))
100×100×100 Array{Gray{Float64},3} with eltype Gray{Float64}:
[:, :, 1] =
```

**Comment by Evizero** [comment]
I think it should mimic the default behaviour when it comes to overall structure and content of the meta text and just replace the `Gray{Float64}(0.4343434)` bits accordingly. 

```
2×16×2 Array{Gray{Float64},3} with eltype Gray{Float64}:
[:, :, 1] =
```

Since 3d-arrays are sliced in the 3d dimension with native julia array printing, we simply do the same for consitency. So I don't think we even have to mentally justify that the 3rd dimension is the most interresting to slice in -- which may or may not be true.

**Comment by Evizero** [comment]
Thank you all for working on this. @johnnychen94 If you do have the time / interesst, I would glady trust your judgment when it comes to merging and tagging the final PR

**Comment by johnnychen94** [comment]
> If you do have the time / interesst, I would glady trust your judgment when it comes to merging and tagging the final PR

Of course I'm interested in this!

**Comment by johnnychen94** [comment]
[MosaicViews](https://github.com/JuliaArrays/MosaicViews.jl) provides another rendering choice, which is more attractive to me. How do you think?

**Comment by mcabbott** [comment]
Or you can `hvcat` things into a matrix yourself. But the overall result then gets scaled, instead of each slice being displayed full-size.

**Comment by johnnychen94** [comment]
Preview:

```julia
julia> Gray.(rand(2,16, 100))
2×16×100 Array{Gray{Float64},3} with eltype Gray{Float64}:
[:, :, 1] =
████████████████████████████████
████████████████████████████████

[:, :, 2] =
████████████████████████████████
████████████████████████████████

[:, :, 3] =
████████████████████████████████
████████████████████████████████

...

[:, :, 98] =
████████████████████████████████
████████████████████████████████

[:, :, 99] =
████████████████████████████████
████████████████████████████████

[:, :, 100] =
████████████████████████████████
████████████████████████████████
```

File: src/imshow.jl [hunk 1]
```diff
@@ -9,6 +9,15 @@ If working in the REPL, the function tries to choose the encoding
 based on the current display size. The image will also be
 downsampled to fit into the display (using `restrict`).
 """
+function imshow(
+        io::IO,
+        img::AbstractArray{<:Colorant, 3},
```
Comment (mcabbott) [inline src/imshow.jl]:
```suggestion
        img::AbstractArray{<:Colorant},
```
Why not just match the `show` method & allow any N here? 1 & 2 should go to more specific methods anyway

Comment (johnnychen94) [reply src/imshow.jl]: Thanks for pointing it out. Now it also works for nd `imshow`

```julia
julia> Gray.(rand(2, 16, 2, 2))
2×16×2×2 Array{Gray{Float64},4} with eltype Gray{Float64}:
[:, :, 1, 1] =
████████████████████████████████
████████████████████████████████

[:, :, 2, 1] =
████████████████████████████████
████████████████████████████████

[:, :, 1, 2] =
████████████████████████████████
████████████████████████████████

[:, :, 2, 2] =
████████████████████████████████
████████████████████████████████
```

**Comment by johnnychen94** [comment]
I believe this is ready to be merged and tagged @Evizero

**Review by Evizero — APPROVED** [review]
LGTM! thanks

**Pull request merged**